src/app/deposit/deposit-dialog/deposit-dialog.component.ts

```diff
@@ -108,12 +108,25 @@ export class DepositDialogComponent implements OnInit {
    * @private
    */
   private adjustDepositAmount(amount: StablecoinBigNumber, min?: StablecoinBigNumber): StablecoinBigNumber {
-    if (!min) return amount
+    if (!min || min.lt(constants.Zero)) return amount
 
-    const safetyMargin = this.conversion.toStablecoin('1')
-    const shouldAdjust = amount.lt(min.add(safetyMargin))
+    const minSafe = this.addSafetyMargin(min)
```

Comment:
Maybe just use scaled value for `minSafe`.
```suggestion
    const minSafe = this.conversion.scale(min, 1.0005)
```

Reply:
Yeah, that would be more precise.